internal/provision/profiles.go

```diff
@@ -34,81 +38,157 @@ const (
-				continue
-			}
-		} else {
+	if len(bytes) == 0 {
+		return nil, nil
+	}
+
+	var files []string
+	err = json.Unmarshal(bytes, &files)
+	if err != nil {
+		return nil, errors.NewCommonEdgeX(errors.KindServerError, "could not unmarshal Profile list contents", err)
+	}
+	if len(files) == 0 {
+		return nil, nil
+	}
+
+	baseUrl, _ := path.Split(inputURI)
+	lc.Infof("Loading pre-defined profiles from %s(%d files found)", displayURI, len(files))
+	var addProfilesReq, processedProfilesReq []requests.DeviceProfileRequest
+	for _, file := range files {
+		fullPath, parsedFullPath := GetFullAndParsedURI(baseUrl, file, "profile", lc)
+		if fullPath == "" || parsedFullPath == nil {
 			continue
 		}
 
-		res, err := dpc.DeviceProfileByName(context.Background(), profile.Name)
-		if err == nil {
-			lc.Infof("Profile %s exists, using the existing one", profile.Name)
-			_, exist := cache.Profiles().ForName(profile.Name)
-			if !exist {
-				err = cache.Profiles().Add(dtos.ToDeviceProfileModel(res.Profile))
-				if err != nil {
-					return errors.NewCommonEdgeX(errors.KindServerError, fmt.Sprintf("failed to cache the profile %s", res.Profile.Name), err)
-				}
-			}
-		} else {
-			lc.Infof("Profile %s not found in Metadata, adding it ...", profile.Name)
-			req := requests.NewDeviceProfileRequest(profile)
-			addProfilesReq = append(addProfilesReq, req)
+		processedProfilesReq, edgexErr = processProfiles(fullPath, parsedFullPath.Redacted(), secretProvider, lc, dpc)
+		if edgexErr != nil {
+			return nil, edgexErr
+		}
+		if len(processedProfilesReq) > 0 {
+			addProfilesReq = append(addProfilesReq, processedProfilesReq...)
 		}
 	}
+	return addProfilesReq, nil
+}
 
-	if len(addProfilesReq) == 0 {
-		return nil
+func processProfiles(fullPath, displayPath string, secretProvider bootstrapInterfaces.SecretProvider, lc logger.LoggingClient, dpc interfaces.DeviceProfileClient) ([]requests.DeviceProfileRequest, errors.EdgeX) {
+	var profile dtos.DeviceProfile
+	var addProfilesReq []requests.DeviceProfileRequest
+
+	fileType := GetFileType(fullPath)
+
+	// if the file type is not yaml or json, it cannot be parsed - just return to not break the loop for other devices
+	if fileType == OTHER {
+		return nil, nil
 	}
-	ctx := context.WithValue(context.Background(), common.CorrelationHeader, uuid.NewString()) // nolint:staticcheck
-	_, edgexErr := dpc.Add(ctx, addProfilesReq)
-	return edgexErr
+
+	content, err := file.Load(fullPath, secretProvider, lc)
+	if err != nil {
+		lc.Errorf("Failed to read Profiles from %s: %v", displayPath, err)
+		return nil, nil
+	}
+
+	switch fileType {
+	case YAML:
+		err = yaml.Unmarshal(content, &profile)
+		if err != nil {
+			lc.Errorf("Failed to YAML decode device profile from %s: %v", displayPath, err)
+			return nil, nil
+		}
+	case JSON:
+		err = json.Unmarshal(content, &profile)
+		if err != nil {
+			lc.Errorf("Failed to JSON decode device profile from %s: %v", displayPath, err)
+			return nil, nil
```

Comment:
@cloudxxx8 , the current code does a `continue` on these error cases. So the service does fail start-up if in these cases.
https://github.com/edgexfoundry/go-mod-bootstrap/blob/main/bootstrap/bootstrap.go#L99-L108

But it does return an error if the call to Core Metadata Add fails (below). 
https://github.com/edgexfoundry/device-sdk-go/blob/main/internal/provision/profiles.go#L96-L99

Shouldn't the error in loading the files also cause bootstrapping to fail also?